pulp_container/app/authorization.py

```diff
@@ -92,32 +118,70 @@ def _split_into_encoded_groups(self, encoded_base32):
                 kid += char
         return kid
 
-    def determine_access(self, user, scope):
+    def determine_access(self):
         """
         Determine access permissions for a corresponding user.
 
         This method determines whether the user has a valid access permission or not.
-        The determination is based on role based access control. For now, the access
-        is given out to anybody because the role based access control is not implemented
-        yet.
-
-        Args:
-            user (str): A name of the user who is trying to access a registry.
-            scope (str): A requested scope.
+        The determination is based on role based access control.
 
         Returns:
             list: An intersected set of the requested and the allowed access.
 
         """
-        typ, name, actions = scope.split(":")
-        actions_list = actions.split(",")
-        permissions = {"pull"}
-        if user == "admin":
-            permissions.add("push")
-        permitted_actions = list(set(actions_list).intersection(permissions))
-        return {"type": typ, "name": name, "actions": permitted_actions}
-
-    def _generate_claim_set(self, issuer, issued_at, subject, audience, access):
+        typ, name, actions = self.scope.split(":")
+        actions = set(actions.split(","))
+
+        permitted_actions = set()
+        for action in actions:
+            has_permission = self.actions_permissions[action](name)
+            if has_permission:
+                permitted_actions.add(action)
+
+        return {"type": typ, "name": name, "actions": list(permitted_actions)}
+
+    def has_pull_permissions(self, path):
+        """
+        Check if the user has permissions to pull from the repository specified by the path.
+        """
+        if self.user.has_perm("container.view_containerdistribution"):
+            return True
+
+        try:
+            distribution = ContainerDistribution.objects.get(base_path=path)
+        except ContainerDistribution.DoesNotExist:
+            return False
+        if self.user.has_perm("container.view_containerdistribution", distribution):
+            return True
+
+        return False
+
+    def has_push_permissions(self, path):
+        """
+        Check if the user has permissions to push to the repository specified by the path.
+        """
+        if not self.user.has_perm("container.add_containerdistribution"):
+            return False
+
+        namespace = path.split("/")[0]
+        return NamespacePermissionsChecker.has_permissions(
+            namespace, self.user, "container.manage_namespace_distributions"
+        )
+
+    def has_view_catalog_permissions(self, path):
+        """
+        Check if the authenticated user is an administrator and is requesting the catalog endpoint.
+        """
+        if path == "catalog" and self.user.username == "admin":
```

Comment:
https://github.com/pulp/pulp_container/pull/203#discussion_r557514184


maybe allow "self.user.is_staff()"
But should we not have a more open restriction here?
Should we do permission based queryset filtering?
@mdellweg

Reply:
The requirement was to allow only the administrator to perform such an action.

Regarding the permission based queryset filtering, I am not sure whether this is going to be resolved in a separate ticket here https://pulp.plan.io/issues/8068.

Reply:
:+1: other ticket seems fine.

Reply:
I wanted to check-in with AH first to understand what are their usecases. Docker hub seems to to deny access to this endpoint unless you have admin perms or maybe this is granted to specific users. I was not able to access this enpoint whether i was anonymous user or authed.
here is the follow up story https://pulp.plan.io/issues/8068 but it needs to be confirmed